### PR TITLE
Handle wasmtime and wasmedge exit status

### PR DESCRIPTION
**Pull request by jprendes**
This PR returns the actual exit code retirned by the wasm module rather than `0` on success and `137` on failre.

### CODE REVIEW

File: crates/containerd-shim-wasmtime/src/instance/instance_linux.rs [hunk 1]
```diff
@@ -120,7 +121,7 @@ mod wasitest {
             ;; Import the required fd_write WASI function which will write the given io vectors to stdout
             ;; The function signature for fd_write is:
             ;; (File Descriptor, *iovs, iovs_len, nwritten) -> Returns number of bytes written
-            (import "wasi_unstable" "fd_write" (func $fd_write (param i32 i32 i32 i32) (result i32)))
+            (import "wasi_snapshot_preview1" "fd_write" (func $fd_write (param i32 i32 i32 i32) (result i32)))
```
Comment (Mossaka) [inline crates/containerd-shim-wasmtime/src/instance/instance_linux.rs:124]: oh nice!

File: crates/containerd-shim-wasmtime/src/executor.rs [hunk 2]
```diff
@@ -39,10 +39,14 @@ impl Executor for WasmtimeExecutor {
...
                 log::info!("calling start function");
-                match f.call(&mut store, &[], &mut []) {
-                    Ok(_) => std::process::exit(0),
-                    Err(_) => std::process::exit(137),
-                };
+
+                let status = f.call(&mut store, &[], &mut []);
+                let status = status
+                    .map(|_| 0)
+                    .map_err(maybe_exit_on_error)
```
Comment (Mossaka) [inline crates/containerd-shim-wasmtime/src/executor.rs:46]: I love this

**Review by Mossaka — APPROVED** [review]
lgtm! thanks

**Comment by Mossaka** [comment]
Could you please take a look @jsturtevant ?

**Comment by jsturtevant** [comment]
LGTM

**Pull request merged**